src/python_x509_pkcs11/lib.py

```diff
@@ -1,66 +1,88 @@
 """Module which have common functions and constants"""
 
-from typing import Dict, List
+from enum import Enum
+from typing import Dict, Set
 
 from asn1crypto.algos import SignedDigestAlgorithm, SignedDigestAlgorithmId
 from pkcs11 import KeyType
 
 DEBUG = False
 
-# Key types and sizes we support
-KEY_TYPES: List[str] = [
-    "ed25519",
-    "ed448",
-    "secp256r1",
-    "secp384r1",
-    "secp521r1",
-    "rsa_2048",
-    "rsa_4096",
-]
-
-KEY_TYPE_VALUES: Dict[str, KeyType] = {
-    "ed25519": KeyType.EC_EDWARDS,
-    "ed448": KeyType.EC_EDWARDS,
-    "secp256r1": KeyType.EC,
-    "secp384r1": KeyType.EC,
-    "secp521r1": KeyType.EC,
-    "rsa_2048": KeyType.RSA,
-    "rsa_4096": KeyType.RSA,
+
+# Enum for Key types and sizes we support
+class KEYTYPES(Enum):
+    ED25519 = "ed25519"
+    ED448 = "ed448"
+    SECP256r1 = "secp256r1"
+    SECP384r1 = "secp384r1"
+    SECP521r1 = "secp521r1"
+    RSA2048 = "rsa_2048"
+    RSA4096 = "rsa_4096"
+
+
+# This is the default key type
+DEFAULT_KEY_TYPE = KEYTYPES.ED25519
+
+KEY_TYPE_VALUES: Dict[KEYTYPES, KeyType] = {
+    KEYTYPES.ED25519: KeyType.EC_EDWARDS,
+    KEYTYPES.ED448: KeyType.EC_EDWARDS,
+    KEYTYPES.SECP256r1: KeyType.EC,
+    KEYTYPES.SECP384r1: KeyType.EC,
+    KEYTYPES.SECP521r1: KeyType.EC,
+    KEYTYPES.RSA2048: KeyType.RSA,
+    KEYTYPES.RSA4096: KeyType.RSA,
 }
 
 
-def signed_digest_algo(key_type: str) -> SignedDigestAlgorithm:
+def get_keytypes_enum(value: str) -> KEYTYPES:
```

Comment:
Do we really need this function? It would be nice to just use the enum when using this lib, a modern IDE will even help with the available options.

Reply:
If we remove the `str` style argument completely and then only use `KEYTYPES` as function argument, there this function will be really helpful for any user of the library. People will have the type as string from other libraries or certificates/configuration itself, and then they can use this function to get the correct `Enum` without repeating the same `if/else` code. 

I am going to export the usable functions and types from the module's `__init__` itself and also heavily document it to make the life easier in future.

Reply:
That's was my point, but I did draw an other conclusion :) 
I understand the need to be plug-able between different libs and so forth. But since it's just a string I believe the comparison would make a bit more sense on the businesses side. 

But, since you wrote it, I believe you have thinking this though more than I have :)

Reply:
> But since it's just a string I believe the comparison would make a bit more sense on the businesses side.

Yes, but also makes a lot of code repeat in the user side, just having a utility function from our side helps in those cases.